pkg/ipam/multipool.go

```diff
@@ -552,7 +552,14 @@ func (m *multiPoolManager) updateLocalNode(ctx context.Context) error {
 		cidrs := []types.IPAMPodCIDR{}
 		if v4Pool := pool.v4; v4Pool != nil {
 			if m.isRestoreFinishedLocked(IPv4) {
-				v4Pool.releaseExcessCIDRsMultiPool(neededIPs.IPv4Addrs)
+				// releaseExcessCIDRsMultiPool interprets neededIPs as how many
+				// free addresses must remain after a CIDR is dropped.
+				// Therefore we subtract the number of in-use addresses from neededIPs.
+				freeNeeded4 := neededIPs.IPv4Addrs - v4Pool.inUseIPCount()
+				if freeNeeded4 < 0 {
+					freeNeeded4 = 0
+				}
```

Comment:
nit: this could be shortened to

```suggestion
				freeNeeded4 := max(neededIPs.IPv4Addrs - v4Pool.inUseIPCount(), 0)
```

Reply:
Yeah that's a lot cleaner thanks. I made the change and force pushed before I saw I could apply the suggestion from here, so it cancelled your auto-merge - sorry about that.

Reply:
No worries. I've re-triggered all tests and re-enabled auto-merge. Thanks again for your contribution!

Reply:
@alimehrabikoshki looks like another change is needed to please the Go formatting linter: https://github.com/cilium/cilium/actions/runs/16370115570/job/46258581876?pr=40393 (running `gofmt -w pkg/ipam/multipool.go` should do).

Reply:
ahhh sorry, I keep forgetting to run it. Really need to integrate fmt with my git workflow. Should be done now. Thanks for the review 👍